### PR TITLE
Bug/unittests

### PR DESCRIPTION
**Pull request by benureau**

### CODE REVIEW

File: pubs/content.py [hunk 1]
```diff
@@ -70,9 +70,14 @@ def read_text_file(filepath, fail=True):
     try:
         with _open(filepath, 'r') as f:
             content = f.read()
+            try:  # Python 2
```
Comment (omangin) [inline pubs/content.py]: We could move this one to the `_open` function so that all python 2 stuff is handled there. It would return a `bytestring` only when `'b'` is in the `mode` argument, regardless of python version.

File: pubs/content.py [hunk 2]
```diff
@@ -47,12 +48,11 @@ def system_path(path):
 
 
 def _open(path, mode):
-    if 'b' in mode:
+    if 'b' in mode or sys.version_info < (3,):
```
Comment (omangin) [inline pubs/content.py]: Should we move this one to the `p3` module?

File: pubs/uis.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 from .content import check_file, read_text_file, write_file, system_path
 
 
+DEBUG = False
```
Comment (omangin) [inline pubs/uis.py]: Do we need to keep this one?

File: pubs/uis.py [hunk 4]
```diff
@@ -117,7 +118,7 @@ def handle_exception(self, exc):
 
         :returns: True if exception has been handled (currently never happens)
         """
-        if not self.debug:
+        if (not DEBUG) and (not self.debug):
```
Comment (omangin) [inline pubs/uis.py]: Same as above.

File: setup.py [hunk 5]
```diff
@@ -19,7 +19,8 @@
                             'pubs.plugs', 'pubs.plugs.alias'],
     scripts      = ['pubs/pubs'],
 
-    install_requires = ['pyyaml', 'bibtexparser', 'python-dateutil', 'requests', 'configobj',
+    install_requires = ['pyyaml', 'bibtexparser', 'python-dateutil', 'requests',
+                        'configobj', 'argcomplete',
```
Comment (omangin) [inline setup.py]: (This commit should not be part of this PR.)
Anyway, since the auto-completion still requires manual configuration I don't believe it worth it to include it as a requirement. We should however definitely add it as an `extras_require`.

File: tests/fake_env.py [hunk 6]
```diff
@@ -5,8 +5,11 @@
 import glob
 
 import dotdot
-from pyfakefs import (fake_filesystem, fake_filesystem_shutil,
-                      fake_filesystem_glob, fake_filesystem_unittest)
+
+from pyfakefs import fake_filesystem, fake_filesystem_unittest
+
+# from pyfakefs import (fake_filesystem, fake_filesystem_shutil,
+#                       fake_filesystem_glob, fake_filesystem_unittest)
```
Comment (omangin) [inline tests/fake_env.py]: The commented lines can be removed.

File: tests/test_usecase.py [hunk 7]
```diff
@@ -124,7 +124,7 @@ def execute_cmds(self, cmds, capture_output=CAPTURE_OUTPUT):
                     self.fail('Unexpected input asked by command: {}.'.format(
                         actual_cmd))
             if PRINT_OUTPUT:
-                print(outs)
+               print(outs)
```
Comment (omangin) [inline tests/test_usecase.py]: Missing space

**Comment by omangin** [comment]
So, I removed the commit with the requirements and fixed the indentation, an unused import, and the commented code.

Feel free to remove the `DEBUG` in `uis.py` Otherwise, the code is good to merge for me!

File: pubs/uis.py (hunk 3 above)
Comment (benureau) [inline pubs/uis.py]: Yep. There is no easy way to make exceptions verbose in the usecase tests, since it involves changing the configuration file. We need a way to make gui error message into true exception when debugging test: this is a simple way to do that.

File: pubs/content.py (hunk 2 above)
Comment (benureau) [inline pubs/content.py]: I don't know if that would make it any clearer. If you have an idea, I am all hear.

**Comment by benureau** [comment]
@omangin Thanks for the review and the changes! Merging.
